blacksmith/src/lib.rs

```diff
@@ -45,6 +45,7 @@ pub struct Blacksmith {
     rustup: Vec<String>,
     stable_version: Option<String>,
     platforms: BTreeMap<String, Platform>,
+    previous_stable_versions: Vec<(String, Vec<String>)>,
```

Comment:
This causes an error if someone has a pre-existing cache file. Would it be possible to add `#[serde(default)]` or some other solution to avoid that?

Reply:
Thanks for catching that. I didn't think to check cache behavior.